File: practical02/Conversion.c
#include <stdio.h>
#include <math.h>

int main(void) {

/* Declare variables */
   int i,inum,tmp,numdigits;
   float fnum;
   char binnum[60];

/* Intialise 4-byte integer */
   inum = 33554431;
/* Convert to 4-byte float */
   fnum = (float) inum;


/* Convert to binary number (string)*/
   i = 0; tmp = inum;
   while (tmp > 0) {
     sprintf(&binnum[i],"%1d",tmp%2);
     tmp = tmp/2;
     i++;
   }

/* Terminate the string */
   binnum[i] = '\0';
       

/* Complete the expression */
   i=0;
   numdigits=fnum;
   while(numdigits>0){
    numdigits=numdigits/10;
    i=i+1;
   }
   numdigits=i;

   printf("inum=%d,  fnum=%f, inum in binary=%s\n",
      inum,fnum,binnum);
   printf("The number of digits is %d\n",numdigits);
}

File: practical02/Conversion.f90
program conversion
! Declare variables 
   integer (kind=4) :: i,inum,tmp,numdigits
   integer :: binnum(60)
   real(kind=4) :: fnum

! Intialise 4-byte integer
   inum = 33554431
! Convert to 4-byte float
   fnum = real(inum)       

! Binary converter using modulus   
   i=1
   tmp=inum
   do while (tmp > 0)
      if(mod(tmp,2)==0) then
        binnum(i) = 0
        else
             binnum(i) = 1
      end if
      tmp = tmp/2
      i = i+1
   end do

   i=0
   numdigits = nint(fnum)
   do while (numdigits>0)
      numdigits=numdigits/10
      i=i+1
   end do
   numdigits=i

   write(6,'(a,i0,a,f0.1,a,b0,/,a,i1)')  'inum=',inum,', fnum=',fnum,', inum in binary=',inum,'numdigits=',numdigits


end program conversion
